### DISCUSSION TITLE
HTML conformance (validation)

### DISCUSSION THREAD

**Discussion by j9t** [Devographics/surveys#161] [State of HTML Design]
Ask about HTML conformance:

* What do developers think about HTML conformance/validation?
* What is being used for conformance testing, i.e. validation?
* If code is not being validated, why not?

**Comment by LeaVerou** [comment]
That (and your other 2 threads) is fantastic input! I moved it to a different category however, as I think it's a better fit for Part 2 of the survey, i.e. not for the list of features with the 3 specific answers.

**Comment by LeaVerou** [comment]
I’m not sure what the value is in seeing what specific validators are used (but it's entirely possible I’m missing it). However, I do think there’s value in seeing how important valid HTML is for authors, as I’ve been in many design discussions that were limited by the lack of data on this.

It would be good to also capture the "It depends" answers, somehow. E.g. there are two conflicting ideas, both reasonably widespread:
1. That many developers want to send valid HTML down the wire, but don't mind invalid HTML that's constructed at runtime. 
2. There is also widespread belief for the exact opposite! E.g. VueJS advertises (or used to) that it produces valid markup at runtime, even though it requires invalid attributes as part of its template syntax.

I also suspect it may vary depending on the type of error, e.g. invalid attribute names may be more easily tolerated than e.g. incorrectly nested elements.

We could keep it simple with:

> How much does valid markup matter to you?
> - A lot, I use a validator and do not tolerate any invalid markup
> - Somewhat / it depends
> - Not at all

With added nuance via [quick context](https://github.com/Devographics/surveys/discussions/163), if the second option is selected, e.g.:
- Easy to forget
- Invalid attribute names ok
- Invalid markup ok at runtime
- Invalid markup ok in source

**Reply by j9t** [reply]
I like the idea of probing for what validating may depend on. Is free input an option?

Otherwise some entirely different factors might be at play that may or may not be captured—e.g., it could depend on the company or a customer, or whether there is enough time, or the engineering culture in place, etc.

**Reply by j9t** [reply]
(Missed to mention, @LeaVerou, definitely down to generating more questions and to helping fine-tune them. Will try to engage here in GitHub Discussions, but if that doesn’t work well, know that I may be able to contribute where deemed useful.)

**Reply by LeaVerou** [reply]
It *is* a free input, these are just quick responses that can serve as a starting point, see the linked discussion.

**Reply by j9t** [reply]
Ah! Missed that :)
